lib/pf/violation.pm

```diff
@@ -476,7 +476,7 @@ sub violation_add {
 
         # check if we are under the grace period of a previous violation
         my ($remaining_time) = violation_grace( $mac, $vid );
-        if ( $remaining_time > 0 ) {
+        if ( $remaining_time > 0 && $data{'force'} ne $TRUE ) {
```

Comment:
Maybe a detail but something like that should be usefull:
} elsif ( $remaining_time > 0 && $data{'force'} eq $TRUE ) {
            my $msg = "Force violation $vid for node $mac";
            $logger->info($msg);
} else { ...